lib/SILGen/SILGen.cpp

```diff
@@ -1280,6 +1280,28 @@ void SILGenModule::postEmitFunction(SILDeclRef constant,
   F->verifyIncompleteOSSA();
 
   emitDifferentiabilityWitnessesForFunction(constant, F);
+
+  // To support using metatypes as type hints in Embedded Swift. A default
+  // argument generator might be returning a metatype, which we normally don't
+  // support in Embedded Swift, but to still allow metatypes as type hints, we
+  // make the generator always inline to the callee by marking it transparent.
+  if (M.getOptions().EmbeddedSwift) {
+    if (constant.isDefaultArgGenerator()) {
```

Comment:
Please do this in SILDeclRef::isTransparent() instead

Reply:
Done